src/CMakeLists.txt

```diff
@@ -1324,53 +1324,42 @@ add_dependencies(cvc5-obj cvc5base cvc5context)
 # Add libcvc5 dependencies for generated sources.
 add_dependencies(cvc5-obj gen-expr gen-versioninfo gen-options gen-tags gen-theory)
 
+if(ENABLE_STATIC_BUILD)
+  add_library(cvc5 STATIC $<TARGET_OBJECTS:cvc5-obj> $<TARGET_OBJECTS:cvc5base> $<TARGET_OBJECTS:cvc5context>)
+else()
+  add_library(cvc5 SHARED $<TARGET_OBJECTS:cvc5-obj> $<TARGET_OBJECTS:cvc5base> $<TARGET_OBJECTS:cvc5context>)
+endif()
```

Comment:
I'm currently wondering why we don't use `BUILD_SHARED_LIBS`? If we would we wouldn't need this if/else and
```
add_library(cvc5 $<TARGET_OBJECTS:cvc5-obj> $<TARGET_OBJECTS:cvc5base> $<TARGET_OBJECTS:cvc5context>)
```
would suffice.

Reply:
Oh. I was not aware of this mechanism. Yes, we should use it!

Reply:
Okay, done.